lib/git_local/repository.rb

```diff
@@ -104,5 +109,10 @@ def reset_to_latest_from_origin
     def repo_path
       @repo_path ||= "#{local_directory}/#{org_repo}"
     end
+
+    def valid_protocol(protocol)
```

Comment:
It feels a little odd to be both validating and returning here. What if this was `validate_protocol`, and we simply did the assignment in the consume case elsewhere?